src/coreComponents/physicsSolvers/SolverBase.cpp

```diff
@@ -604,6 +586,51 @@ bool SolverBase::lineSearchWithParabolicInterpolation( real64 const & time_n,
   return lineSearchSuccess;
 }
 
+real64 SolverBase::assembleResidualOnlyAndCalculateNorm( real64 const & time_n,
```

Comment:
Is the goal of combining two functions to allow solvers to fuse the two kernels together?

Reply:
The goal was to eliminate some redundant code in three places where we recompute the residual and calculate the norm (2 line search functions and the sequential convergence check). 

But, in the last commit I reverted my change to `SolverBase` and removed this function. I think it was odd to have this function not fully implemented (by not fully implemented, I mean still assembling the Jacobian). We can bring it back later when we have functions computing the residual only.